modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/config/ClientConnFactoryBuilder.java

```diff
@@ -341,15 +349,120 @@ private SSLContext createSSLContext(OMElement keyStoreElt, OMElement trustStoreE
+        }
+    }
+
+    private SSLContext createSSLContext(OMElement keyStoreElt, OMElement trustStoreElt,
+                                        boolean novalidatecert, SecretResolver secretResolver) throws AxisFault {
+
+        KeyManager[] keymanagers = null;
+        TrustManager[] trustManagers = null;
+
+
+        if (keyStoreElt != null) {
+            String location = keyStoreElt.getFirstChildWithName(new QName("Location")).getText();
+            String type = keyStoreElt.getFirstChildWithName(new QName("Type")).getText();
+            String storePassword = getSecureVaultValue(secretResolver, keyStoreElt.getFirstChildWithName(new QName
+                    ("Password")));
+            String keyPassword = getSecureVaultValue(secretResolver, keyStoreElt.getFirstChildWithName(new QName
+                    ("KeyPassword")));
+
+            FileInputStream fis = null;
+            try {
+                KeyStore keyStore = KeyStore.getInstance(type);
+                fis = new FileInputStream(location);
+                if (log.isInfoEnabled()) {
+                    log.info(name + " Loading Identity Keystore from : " + location);
+                }
+
+                keyStore.load(fis, storePassword.toCharArray());
+                KeyManagerFactory kmfactory = KeyManagerFactory.getInstance(
+                        KeyManagerFactory.getDefaultAlgorithm());
+                kmfactory.init(keyStore, keyPassword.toCharArray());
+                keymanagers = kmfactory.getKeyManagers();
+
+            } catch (GeneralSecurityException gse) {
+                log.error(name + " Error loading Keystore : " + location, gse);
+                throw new AxisFault("Error loading Keystore : " + location, gse);
+            } catch (IOException ioe) {
+                log.error(name + " Error opening Keystore : " + location, ioe);
+                throw new AxisFault("Error opening Keystore : " + location, ioe);
+            } finally {
+                if (fis != null) {
+                    try {
+                        fis.close();
+                    } catch (IOException ignore) {
+                    }
+                }
+            }
+        }
+
+        if (trustStoreElt != null) {
+            if (novalidatecert && log.isWarnEnabled()) {
+                log.warn(name + " Ignoring novalidatecert parameter since a truststore has been specified");
+            }
+
+            String location = trustStoreElt.getFirstChildWithName(new QName("Location")).getText();
+            String type = trustStoreElt.getFirstChildWithName(new QName("Type")).getText();
+            String storePassword = getSecureVaultValue(secretResolver, trustStoreElt.getFirstChildWithName(new QName
+                    ("Password")));
+
+            FileInputStream fis = null;
+            try {
+                KeyStore trustStore = KeyStore.getInstance(type);
+                fis = new FileInputStream(location);
```

Comment:
Shall we use [1]
[1] https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
since the input stream is closed correctly, I think my original code would be fine as well